internal/applications/applications.go

```diff
@@ -0,0 +1,99 @@
+package applications
+
+import (
+	"reflect"
+	"sort"
+
+	"github.com/vshn/appcat-cli/internal/defaults"
+	"github.com/vshn/appcat-cli/internal/util"
+	exoscalev1 "github.com/vshn/component-appcat/apis/exoscale/v1"
+	vshnv1 "github.com/vshn/component-appcat/apis/vshn/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	ExoscaleApiVersion = "exoscale.appcat.vshn.io/v1"
+	VshnApiVersion     = "vshn.appcat.vshn.io/v1"
+)
+
+var (
+	// List of supported apps
+	//
+	// This is mostly used to generate `AppMap` (via `MakeAppMap`).
+	//
+	// In order to avoid allocations of (potentially) large types, pass in a nil pointer (`(*type)(nil)`) to `NewApp`.
+	Apps = []App{
+		// Exoscale
+		NewApp(ExoscaleApiVersion, "ExoscalePostgreSQL", (*exoscalev1.ExoscalePostgreSQLSpec)(nil)),
+		NewApp(ExoscaleApiVersion, "ExoscaleRedis", (*exoscalev1.ExoscaleRedisSpec)(nil)),
+		NewApp(ExoscaleApiVersion, "ExoscaleKafka", (*exoscalev1.ExoscaleKafkaSpec)(nil)),
+		NewApp(ExoscaleApiVersion, "ExoscaleMySQL", (*exoscalev1.ExoscaleMySQLSpec)(nil)),
+		NewApp(ExoscaleApiVersion, "ExoscaleOpenSearch", (*exoscalev1.ExoscaleOpenSearchSpec)(nil)),
+
+		// VSHN
+		NewApp(VshnApiVersion, "VSHNPostgreSQL", (*vshnv1.VSHNPostgreSQLSpec)(nil)),
+		NewApp(VshnApiVersion, "VSHNRedis", (*vshnv1.VSHNRedisSpec)(nil)),
+	}
+)
+
+// App describes a supported AppCat application.
+//
+// The data in here is used to generate command-line help, and some scaffolding
+// around instantiating new applications,
+type App struct {
+	// Metadata "template" used to instantiate new objects as well as for lookup
+	metav1.TypeMeta
+	spec reflect.Type
```

Comment:
Do we need this to be of type reflect.Type ?
In the case where we want to refactor the field setting and field lookups in the future to something else than the reflect package the refactor would be more extensive.

Reply:
What would you do instead?